### PR TITLE
[SOAR-3727] - Updated Rest plugin to accept blank bodies

### PR DESCRIPTION
**Pull request by jmcadams-r7**
## Description: 
This PR fixes an issue where if a successful request returned a null body, the plugin would crash. It will now return a blank body object. 

## Validation: 
NOTE: Validators crashed
```
RMT-MBP-5357:rest jmcadams$ icon-validate .
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
Traceback (most recent call last):
  File "/usr/local/bin/icon-validate", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/icon_validator/__main__.py", line 52, in main
    validate(directory=path, spec_file_name=spec_file_name)
  File "/usr/local/lib/python3.8/site-packages/icon_validator/validate.py", line 35, in validate
    v.validate(spec)
  File "/usr/local/lib/python3.8/site-packages/icon_validator/rules/plugin_validators/output_validator.py", line 44, in validate
    schemas = OutputValidator.get_schemas(spec)
  File "/usr/local/lib/python3.8/site-packages/icon_validator/rules/plugin_validators/output_validator.py", line 31, in get_schemas
    schemas[os.path.basename(path)] = OutputValidator.read_schema(full_path)
  File "/usr/local/lib/python3.8/site-packages/icon_validator/rules/plugin_validators/output_validator.py", line 40, in read_schema
    json_ = json.loads(re.findall(output_pattern, text)[1])  # 0 for input, 1 for output
IndexError: list index out of range
```

## Testing: 
```
RMT-MBP-5357:rest jmcadams$ ./run.sh -R all -j
Running: cat tests/delete.json | docker run --rm   -i rapid7/rest:3.0.4  run | grep -- ^\{ | jq -r '.body | try(.log | split("\n") | .[]),.output'
Connect: Configuring REST details
Connect: Connecting..
rapid7/REST:3.0.3. Step name: delete
Starting new HTTPS connection (1): api.nasa.gov:443
/usr/local/lib/python3.7/site-packages/urllib3-1.25.3-py3.7.egg/urllib3/connectionpool.py:851: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
https://api.nasa.gov:443 "DELETE /planetary/earth/imagery/?lat=1.5&lon=100.75&date=2014-02-01&cloud_score=True&api_key=DEMO_KEY HTTP/1.1" 405 178
Connect: Configuring REST details
Connect: Connecting..
rapid7/REST:3.0.3. Step name: delete

{
  "body_object": {},
  "body_string": "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 3.2 Final//EN\">\n<title>405 Method Not Allowed</title>\n<h1>Method Not Allowed</h1>\n<p>The method is not allowed for the requested URL.</p>\n",
  "status": 405,
  "headers": {
    "Date": "Wed, 02 Dec 2020 16:46:34 GMT",
    "Content-Type": "text/html; charset=utf-8",
    "Content-Length": "178",
    "Connection": "keep-alive",
    "X-RateLimit-Limit": "40",
    "X-RateLimit-Remaining": "23",
    "Allow": "HEAD, OPTIONS, GET",
    "Access-Control-Allow-Origin": "*",
    "Age": "13",
    "Via": "http/1.1 api-umbrella (ApacheTrafficServer [cMs f ])",
    "X-Cache": "MISS",
    "Strict-Transport-Security": "max-age=31536000; preload"
  }
}

Running: cat tests/get.json | docker run --rm   -i rapid7/rest:3.0.4  run | grep -- ^\{ | jq -r '.body | try(.log | split("\n") | .[]),.output'
Connect: Configuring REST details
Connect: Connecting..
rapid7/REST:3.0.3. Step name: get
Starting new HTTPS connection (1): api.nasa.gov:443
/usr/local/lib/python3.7/site-packages/urllib3-1.25.3-py3.7.egg/urllib3/connectionpool.py:851: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
https://api.nasa.gov:443 "GET /planetary/earth/imagery/?lat=1.5&lon=100.75&date=2014-02-01&cloud_score=True&api_key=DEMO_KEY HTTP/1.1" 200 494144
SHIFT_JIS Japanese prober hit error at byte 36
EUC-JP Japanese prober hit error at byte 0
GB2312 Chinese prober hit error at byte 53
EUC-KR Korean prober hit error at byte 0
CP949 Korean prober hit error at byte 35
Big5 Chinese prober hit error at byte 0
EUC-TW Taiwan prober hit error at byte 0
windows-1251 confidence = 0.03538759644666556, below negative shortcut threshhold 0.05
KOI8-R confidence = 0.016963867178444638, below negative shortcut threshhold 0.05
ISO-8859-5 confidence = 0.02844118373173476, below negative shortcut threshhold 0.05
MacCyrillic confidence = 0.03274276410714415, below negative shortcut threshhold 0.05
IBM855 confidence = 0.035830974884709106, below negative shortcut threshhold 0.05
ISO-8859-7 confidence = 0.03751254681234283, below negative shortcut threshhold 0.05
windows-1253 confidence = 0.0375869418405067, below negative shortcut threshhold 0.05
ISO-8859-5 confidence = 0.020896265081302136, below negative shortcut threshhold 0.05
windows-1251 confidence = 0.029499907566232955, below negative shortcut threshhold 0.05
TIS-620 confidence = 0.014616727255389112, below negative shortcut threshhold 0.05
ISO-8859-9 confidence = 0.04162745877016518, below negative shortcut threshhold 0.05
windows-1251 not active
KOI8-R not active
ISO-8859-5 not active
MacCyrillic not active
IBM866 Russian confidence = 0.05522136981525282
IBM855 not active
ISO-8859-7 not active
windows-1253 not active
ISO-8859-5 not active
windows-1251 not active
TIS-620 not active
ISO-8859-9 not active
windows-1255 Hebrew confidence = 0.0
windows-1255 Hebrew confidence = 0.06553460815354639
windows-1255 Hebrew confidence = 0.05939313765106644
no probers hit minimum threshold
utf-8  confidence = 0.010000000000000009
SHIFT_JIS Japanese confidence = 0.01
EUC-JP Japanese confidence = 0.01
GB2312 Chinese confidence = 0.01
EUC-KR Korean confidence = 0.01
CP949 Korean confidence = 0.01
Big5 Chinese confidence = 0.01
EUC-TW Taiwan confidence = 0.01
windows-1251 Russian confidence = 0.03538759644666556
KOI8-R Russian confidence = 0.016963867178444638
ISO-8859-5 Russian confidence = 0.02844118373173476
MacCyrillic Russian confidence = 0.03274276410714415
IBM866 Russian confidence = 0.05522136981525282
IBM855 Russian confidence = 0.035830974884709106
ISO-8859-7 Greek confidence = 0.03751254681234283
windows-1253 Greek confidence = 0.0375869418405067
ISO-8859-5 Bulgairan confidence = 0.020896265081302136
windows-1251 Bulgarian confidence = 0.029499907566232955
TIS-620 Thai confidence = 0.014616727255389112
ISO-8859-9 Turkish confidence = 0.04162745877016518
windows-1255 Hebrew confidence = 0.0
windows-1255 Hebrew confidence = 0.06553460815354639
windows-1255 Hebrew confidence = 0.05939313765106644
windows-1255 Hebrew confidence = 0.05939313765106644
SHIFT_JIS Japanese prober hit error at byte 36
EUC-JP Japanese prober hit error at byte 0
GB2312 Chinese prober hit error at byte 53
EUC-KR Korean prober hit error at byte 0
CP949 Korean prober hit error at byte 35
Big5 Chinese prober hit error at byte 0
EUC-TW Taiwan prober hit error at byte 0
windows-1251 confidence = 0.03538759644666556, below negative shortcut threshhold 0.05
KOI8-R confidence = 0.016963867178444638, below negative shortcut threshhold 0.05
ISO-8859-5 confidence = 0.02844118373173476, below negative shortcut threshhold 0.05
MacCyrillic confidence = 0.03274276410714415, below negative shortcut threshhold 0.05
IBM855 confidence = 0.035830974884709106, below negative shortcut threshhold 0.05
ISO-8859-7 confidence = 0.03751254681234283, below negative shortcut threshhold 0.05
windows-1253 confidence = 0.0375869418405067, below negative shortcut threshhold 0.05
ISO-8859-5 confidence = 0.020896265081302136, below negative shortcut threshhold 0.05
windows-1251 confidence = 0.029499907566232955, below negative shortcut threshhold 0.05
TIS-620 confidence = 0.014616727255389112, below negative shortcut threshhold 0.05
ISO-8859-9 confidence = 0.04162745877016518, below negative shortcut threshhold 0.05
windows-1251 not active
KOI8-R not active
ISO-8859-5 not active
MacCyrillic not active
IBM866 Russian confidence = 0.05522136981525282
IBM855 not active
ISO-8859-7 not active
windows-1253 not active
ISO-8859-5 not active
windows-1251 not active
TIS-620 not active
ISO-8859-9 not active
windows-1255 Hebrew confidence = 0.0
windows-1255 Hebrew confidence = 0.06553460815354639
windows-1255 Hebrew confidence = 0.05939313765106644
no probers hit minimum threshold
utf-8  confidence = 0.010000000000000009
SHIFT_JIS Japanese confidence = 0.01
EUC-JP Japanese confidence = 0.01
GB2312 Chinese confidence = 0.01
EUC-KR Korean confidence = 0.01
CP949 Korean confidence = 0.01
Big5 Chinese confidence = 0.01
EUC-TW Taiwan confidence = 0.01
windows-1251 Russian confidence = 0.03538759644666556
KOI8-R Russian confidence = 0.016963867178444638
ISO-8859-5 Russian confidence = 0.02844118373173476
MacCyrillic Russian confidence = 0.03274276410714415
IBM866 Russian confidence = 0.05522136981525282
IBM855 Russian confidence = 0.035830974884709106
ISO-8859-7 Greek confidence = 0.03751254681234283
windows-1253 Greek confidence = 0.0375869418405067
ISO-8859-5 Bulgairan confidence = 0.020896265081302136
windows-1251 Bulgarian confidence = 0.029499907566232955
TIS-620 Thai confidence = 0.014616727255389112
ISO-8859-9 Turkish confidence = 0.04162745877016518
windows-1255 Hebrew confidence = 0.0
windows-1255 Hebrew confidence = 0.06553460815354639
windows-1255 Hebrew confidence = 0.05939313765106644
windows-1255 Hebrew confidence = 0.05939313765106644
Connect: Configuring REST details
Connect: Connecting..
rapid7/REST:3.0.3. Step name: get

{
  "body_object": {},
  "body_string": <ommited due to size> 
  "status": 200,
  "headers": {
    "Date": "Wed, 02 Dec 2020 16:48:14 GMT",
    "Content-Type": "image/png",
    "Content-Length": "494144",
    "Connection": "keep-alive",
    "X-RateLimit-Limit": "40",
    "X-RateLimit-Remaining": "17",
    "Last-Modified": "Wed, 02 Dec 2020 16:40:29 GMT",
    "Cache-Control": "public, max-age=43200",
    "Expires": "Thu, 03 Dec 2020 04:40:29 GMT",
    "ETag": "\"1606927229.468-494144-1506021431\"",
    "Access-Control-Allow-Origin": "*",
    "Age": "484",
    "Via": "http/1.1 api-umbrella (ApacheTrafficServer [cHs f ])",
    "X-Cache": "HIT",
    "Strict-Transport-Security": "max-age=31536000; preload"
  }
}

Running: cat tests/patch.json | docker run --rm   -i rapid7/rest:3.0.4  run | grep -- ^\{ | jq -r '.body | try(.log | split("\n") | .[]),.output'
Connect: Configuring REST details
Connect: Connecting..
rapid7/REST:3.0.3. Step name: patch
Starting new HTTPS connection (1): api.nasa.gov:443
https://api.nasa.gov:443 "PATCH /planetary/earth/imagery/?lat=1.5&lon=100.75&date=2014-02-01&cloud_score=True&api_key=DEMO_KEY HTTP/1.1" 405 178
Connect: Configuring REST details
Connect: Connecting..
rapid7/REST:3.0.3. Step name: patch

{
  "body_object": {},
  "body_string": "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 3.2 Final//EN\">\n<title>405 Method Not Allowed</title>\n<h1>Method Not Allowed</h1>\n<p>The method is not allowed for the requested URL.</p>\n",
  "status": 405,
  "headers": {
    "Date": "Wed, 02 Dec 2020 16:48:38 GMT",
    "Content-Type": "text/html; charset=utf-8",
    "Content-Length": "178",
    "Connection": "keep-alive",
    "X-RateLimit-Limit": "40",
    "X-RateLimit-Remaining": "16",
    "Allow": "HEAD, OPTIONS, GET",
    "Access-Control-Allow-Origin": "*",
    "Age": "12",
    "Via": "http/1.1 api-umbrella (ApacheTrafficServer [c s f ])",
    "X-Cache": "MISS",
    "Strict-Transport-Security": "max-age=31536000; preload"
  }
}

Running: cat tests/post.json | docker run --rm   -i rapid7/rest:3.0.4  run | grep -- ^\{ | jq -r '.body | try(.log | split("\n") | .[]),.output'
Connect: Configuring REST details
Connect: Connecting..
rapid7/REST:3.0.3. Step name: post
Starting new HTTPS connection (1): api.nasa.gov:443
/usr/local/lib/python3.7/site-packages/urllib3-1.25.3-py3.7.egg/urllib3/connectionpool.py:851: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
https://api.nasa.gov:443 "POST /planetary/earth/imagery/?lat=1.5&lon=100.75&date=2014-02-01&cloud_score=True&api_key=DEMO_KEY HTTP/1.1" 405 178
Connect: Configuring REST details
Connect: Connecting..
rapid7/REST:3.0.3. Step name: post

{
  "body_object": {},
  "body_string": "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 3.2 Final//EN\">\n<title>405 Method Not Allowed</title>\n<h1>Method Not Allowed</h1>\n<p>The method is not allowed for the requested URL.</p>\n",
  "status": 405,
  "headers": {
    "Date": "Wed, 02 Dec 2020 16:48:41 GMT",
    "Content-Type": "text/html; charset=utf-8",
    "Content-Length": "178",
    "Connection": "keep-alive",
    "X-RateLimit-Limit": "40",
    "X-RateLimit-Remaining": "15",
    "Allow": "HEAD, OPTIONS, GET",
    "Access-Control-Allow-Origin": "*",
    "Age": "12",
    "Via": "http/1.1 api-umbrella (ApacheTrafficServer [cMs f ])",
    "X-Cache": "MISS",
    "Strict-Transport-Security": "max-age=31536000; preload"
  }
}

Running: cat tests/put.json | docker run --rm   -i rapid7/rest:3.0.4  run | grep -- ^\{ | jq -r '.body | try(.log | split("\n") | .[]),.output'
Connect: Configuring REST details
Connect: Connecting..
rapid7/REST:3.0.3. Step name: put
Starting new HTTPS connection (1): api.nasa.gov:443
https://api.nasa.gov:443 "PUT /planetary/earth/imagery/?lat=1.5&lon=100.75&date=2014-02-01&cloud_score=True&api_key=DEMO_KEY HTTP/1.1" 405 178
Connect: Configuring REST details
Connect: Connecting..
rapid7/REST:3.0.3. Step name: put

{
  "body_object": {},
  "body_string": "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 3.2 Final//EN\">\n<title>405 Method Not Allowed</title>\n<h1>Method Not Allowed</h1>\n<p>The method is not allowed for the requested URL.</p>\n",
  "status": 405,
  "headers": {
    "Date": "Wed, 02 Dec 2020 16:48:44 GMT",
    "Content-Type": "text/html; charset=utf-8",
    "Content-Length": "178",
    "Connection": "keep-alive",
    "X-RateLimit-Limit": "40",
    "X-RateLimit-Remaining": "14",
    "Allow": "HEAD, OPTIONS, GET",
    "Access-Control-Allow-Origin": "*",
    "Age": "13",
    "Via": "http/1.1 api-umbrella (ApacheTrafficServer [cMs f ])",
    "X-Cache": "MISS",
    "Strict-Transport-Security": "max-age=31536000; preload"
  }
}

```

### CODE REVIEW

**Review by jschipp-r7 — APPROVED** [review]
There is a lot of code duplication here

**Comment by jmcadams-r7** [comment]
I agree with the code duplication comment. However, I think it's out of scope for this bug. If we want to rewrite the plugin for reusability, I'm down.

**Review by mhofert-r7 — APPROVED** [review]

**Review by mhofert-r7 — APPROVED** [review]

File: rest/komand_rest/actions/get/action.py [hunk 1]
```diff
@@ -20,14 +21,18 @@ def run(self, params={}):
...
                                 verify=self.connection.ssl_verify)
         body_object = {}
         try:
             body_object = response.json()
         except ValueError:
             """ Nothing? We don't care if it fails, that could be normal """
+        # It's possible to have a successful call with no body
+        # https://stackoverflow.com/questions/32319845/python-requests-gives-none-response-where-json-data-is-expected
+        if body_object == None:
+            body_object = {}
```
Comment (mrinehart-r7) [inline rest/komand_rest/actions/get/action.py:35]:
```suggestion
            body_object = {"object": {}}
```
Should this match the output structure of a valid body_object?

**Review by ncramer-r7 — APPROVED** [review]

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

File: rest/komand_rest/actions/get/action.py (hunk 1 above)
Comment (jmcadams-r7) [reply rest/komand_rest/actions/get/action.py:35]: I don't think that's valid.

**Review by mrinehart-r7 — APPROVED** [review]

**Pull request merged**